### PR TITLE
grub: enable aarch64-efi support

### PR DESCRIPTION
**Pull request by marmeladema**
grub package does now come without any bootloader by default.
Previously i386-pc bootloader was shipped by default but as it
is not available on aarch64 architecture, it has been separated
in another package called grub-i386-pc. People running x86 pc who
need i386-pc bootloader should install grub-i386-pc package.

efibootmgr still needs to be updated to be available on aarch64.

### CODE REVIEW

File: srcpkgs/grub/template [hunk 1]
```diff
@@ -14,19 +14,29 @@ homepage="https://www.gnu.org/software/grub/"
...
 nopie=yes
 
-subpackages="grub-utils grub-i386-efi"
+subpackages="grub-utils"
+# _SUPPLEMENTARY_BUILDS is a list of <TARGET>-<PLATFORN> version of grub to build
 case "$XBPS_TARGET_MACHINE" in
-	x86_64*) subpackages+=" grub-x86_64-efi"
-		_GRUBX64=1
+	x86_64*)
+		_SUPPLEMENTARY_BUILDS+="i386-pc i386-efi ${XBPS_TARGET_MACHINE}-efi"
```
Comment (maxice8) [inline srcpkgs/grub/template]: if you're appending to a variable put a space before as a safeguard

File: srcpkgs/grub/template [hunk 2]
```diff
@@ -141,19 +147,24 @@ grub-utils_package() {
...
-	}
-}
-grub-i386-efi_package() {
-	noarch=yes
-	depends="grub>=$version dosfstools efibootmgr"
-	short_desc+=" - i386 EFI support"
-	pkg_install() {
-		vmove usr/lib/grub/i386-efi
+
+for _SUPP_BUILD in ${_SUPPLEMENTARY_BUILDS}; do
```
Comment (maxice8) [inline srcpkgs/grub/template]: i'm sorry but what is this ?

why are you generating package() definitions on the fly ?

can't you just keep them all and just use subpackages= to select which ones are necessary ?

Comment (marmeladema) [reply srcpkgs/grub/template]: To avoid duplicating code but as you wish

**Comment by maxice8** [comment]
> 
> To avoid duplicating code but as you wish
> 

It is better to be completely clear and sometimes repeat yourself than to be smart about it.

The future you and other people touching this template will appreciate it.

File: srcpkgs/grub/template [hunk 3]
```diff
@@ -141,19 +148,31 @@ grub-utils_package() {
...
-	short_desc+=" - x86_64 EFI support"
+	short_desc+=" - ${_TARGET} ${_PLATFORM} support"
 	pkg_install() {
-		vmove usr/lib/grub/x86_64-efi
+		vmove usr/lib/grub/${_GRUB_TARGET}-${_PLATFORM}
 	}
 }
+
+grub-i386-pc_package() {
+	supplementary_package i386 i386 pc
```
Comment (maxice8) [inline srcpkgs/grub/template]: please just be explicit and repeat the definition on each package

stop trying to be smart

File: srcpkgs/grub/template [hunk 4]
```diff
@@ -141,19 +148,31 @@ grub-utils_package() {
 		vmove usr/bin/grub-mkfont
 	}
 }
-grub-x86_64-efi_package() {
+
+supplementary_package() {
```
Comment (maxice8) [inline srcpkgs/grub/template]: this is unnecessary and isn't even has the required underscore prefix.

Just repeat the definition on each pacakge.

File: srcpkgs/grub/template (hunk 3 above)
Comment (marmeladema) [reply srcpkgs/grub/template]: You can also stop using such a condescending tone.
I will update the PR to fulfill your desire.

**Comment by marmeladema** [comment]
The real question here, is:
* should the grub package for x86* architectures include the i386-pc platform to be backward compatible ?

**Comment by maxice8** [comment]
Request for review @Gottox

**Comment by marmeladema** [comment]
I tried to enable it for arm 32bit also without luck. Lets keep this PR for aarch64 only. I also silenced the make steps and show the last 100 lines in case of failure so that Travis is happy :)

**Comment by Gottox** [comment]
Is there a platform someone can test aarch64 support with?

**Comment by marmeladema** [comment]
I tested on my roc-rk3328-cc board with success.

**Comment by marmeladema** [comment]
Any news on this @Gottox ?

**Pull request merged**